Repositories/ProjectRepository.cs

```diff
@@ -695,6 +705,30 @@ public async Task SyncProjectToES(Project project)
                      
             
         }
+
+        /// <summary>
+        ///     This method will call the ElasticSearch index to retrieve projects where the title starts with the query.
+        /// </summary>
+        /// <param name="query">The string of characters with which the title must begin</param>
+        /// <returns>
+        ///     This method return a list of projects where the title, or part of the title matches the query.
+        /// </returns>
+        public async Task<List<Project>> FindProjectsWhereTitleStartsWithQuery(string query)
+        {
+            RestRequest request = new RestRequest("_search", Method.POST);
+            string body = queries.SearchProjectsByPartOfTitle
+                .Replace("ReplaceWithQuery", query);
+            request.AddParameter("application/json", body, ParameterType.RequestBody);
+            IRestResponse restResponse = elasticSearchContext.Execute(request);
+            if(!restResponse.IsSuccessful)
+            {
+                throw new Exception(restResponse.ErrorMessage);
+            }
```

Comment:
Here we have to throw another exception (explained earlier). I think, but you should research this a bit more, whenever the rest response is not successful and there is no content, it means that the service is unavailable.